### PR TITLE
Emission

### LINKED ISSUE #39: Emission light
Add emission to the geometry abstract class and add code to consider it in the renderer calculation.

### PR DESCRIPTION
**Pull request by abrahammurciano**
close #39

### CODE REVIEW

File: src/main/primitives/Point.java [hunk 1]
```diff
@@ -105,28 +101,6 @@ public double squareDistance(Point target) {
 		}, target, Point::new).sum();
 	}
 
-	/**
```
Comment (elevin72) [inline src/main/primitives/Point.java:108]: lol

File: src/main/rendering/raytracing/BasicRayTracer.java [hunk 2]
```diff
@@ -24,6 +26,11 @@ public BasicRayTracer(Scene scene) {
 
 	@Override
 	public Colour trace(Ray ray) {
-		return scene.geometries.intersect(ray).isEmpty() ? scene.background : scene.ambient.colour;
+		List<Intersection> intersections = scene.geometries.intersect(ray);
+		if (intersections.isEmpty()) {
+			return scene.background;
+		} else {
+			return scene.ambient.colour.add(ray.closest(intersections).geometry.material.emission);
```
Comment (elevin72) [inline src/main/rendering/raytracing/BasicRayTracer.java:33]: sooooo many dots, Can we add some intermediate functions, Slightly slower, but less ugly.
For example, give geometry a getEmmission() function

Comment (abrahammurciano) [reply src/main/rendering/raytracing/BasicRayTracer.java:33]: i was thinking to put that in intersection. like
```suggestion
			return scene.ambient.colour.add(ray.closest(intersections).emission());
```

Comment (elevin72) [reply src/main/rendering/raytracing/BasicRayTracer.java:33]: :perfect:

**Pull request merged**